hypershift-operator/controllers/hostedcluster/hostedcluster_webhook.go

```diff
@@ -69,6 +74,61 @@ func filterMutableHostedClusterSpecFields(spec *hyperv1.HostedClusterSpec) {
 	}
 }
 
+// validateStructDeepEqual walks through a struct and compares each entry.  If it comes accross a substruct it
+// recursively calls itself.  Returns a list of immutable field errors generated by any field being changed.
+func validateStructDeepEqual(x reflect.Value, y reflect.Value, path *field.Path, errs field.ErrorList) field.ErrorList {
+	for i := 0; i < x.NumField(); i++ {
+		v1 := x.Field(i)
+		v2 := y.Field(i)
+		json_id := x.Type().Field(i).Tag.Get("json")
```

Comment:
golang uses camelCase by convention:
```suggestion
		jsonID := x.Type().Field(i).Tag.Get("json")
```

Also, this means that we will only report the last tag?

Reply:
Should be the first tag..

Reply:
Too many years writing C!! :)